### ISSUE TITLE
use 'Get.put(App())' and 'Get.put<App>(App())' will generate two different instance 

### ISSUE THREAD

**Issue by chejdj** [jonataslaw/getx#2657]
**ATTENTION: DO NOT USE THIS FIELD TO ASK SUPPORT QUESTIONS. USE THE PLATFORM CHANNELS FOR THIS. THIS SPACE IS DEDICATED ONLY FOR BUGS DESCRIPTION.**
**Fill in the template. Issues that do not respect the model will be closed.**

**Describe the bug**
use 'Get.put(App())' and 'Get.put<App>(App())' will register two different instance 
'Get.put(HomeController())' the key is  'HomeController?'
'Get.put<HomeController>(HomeController())'  the key is 'HomeController'

bugs: when using `GetBuilder(init: Get.put(HomeController))` will generate two  instance. One is 'HomeController?' , another instance generates by 'GetBuilderState#isRegistered is false '. 
Because if isRegistered is false, it use this way to registers `GetInstance().put<T>(controller!, tag: widget.tag);`

**Reproduction code
NOTE: THIS IS MANDATORY, IF YOUR ISSUE DOES NOT CONTAIN IT, IT WILL BE CLOSED PRELIMINARY)**

example:

```dart
import 'package:flutter/material.dart';
import 'package:get/get.dart';
import 'package:get/get_rx/get_rx.dart';

void main() => runApp(MaterialApp(home: Home()));

class Home extends StatelessWidget {
  final count = 0.obs;
  @override
  Widget build(context) => Scaffold(
      appBar: AppBar(title: const Text("counter")),
      body: GetBuilder(
        init: Get.put<HomeController>(HomeController()),
        builder: (HomeController controller) {
          return Center(
            child: Obx(() => Text("$count")),
          );
        },
      ),
      floatingActionButton: FloatingActionButton(
        child: const Icon(Icons.add),
        onPressed: () => count.value++,
      ));
}

class HomeController extends GetxController {}
```

**To Reproduce**
1. just run

**Expected behavior**
`Get.put(HomeController())` should generate the key is 'HomeController'

**Screenshots**

**Flutter Version:**
3.3.10

**Getx Version:**
4.6.5

**Describe on which device you found the bug:**
ex: Moto z2 - Android.

**Minimal reproduce code**
Provide a minimum reproduction code for the problem

**Discussion**
guess the Dart Type something error. We should unity the 'put' way

**Comment by chejdj** [comment]
@jonataslaw  Hi, could you see this ?

**Comment by Incro** [comment]
I think the init of the GetBuilder is meant to be used with just the Controller Initialiser without Get.put(). 

This should stop creating multiple controllers:
```dart
class Home extends StatelessWidget {
  final count = 0.obs;
  @override
  Widget build(context) => Scaffold(
      appBar: AppBar(title: const Text("counter")),
      body: GetBuilder(
        init: HomeController(),
        builder: (HomeController controller) {
          return Center(
            child: Obx(() => Text("$count")),
          );
        },
      ),
      floatingActionButton: FloatingActionButton(
        child: const Icon(Icons.add),
        onPressed: () => count.value++,
      ));
}
``` 

If you want to use the HomeController outside of the GetBuilder you can initialise it somewhere else and then do:
```dart
GetBuilder<HomeController>(
          builder: (s) => Text('${s.counter}'),
```
See [https://github.com/jonataslaw/getx/blob/master/documentation/en_US/state_management.md#usage](url)

**Comment by chejdj** [comment]
@Incro Yes, I know the right usage. But I think the `Get.put<HomeController>(HomeController())` and `Get.put(HomeController())` should keep same.

**Comment by kauemurakami** [comment]
to retrieve an already created instance you can just use 
```dart 
final myController = Get.find<YourController>() ?? Get.create(Controller...) ou Get.put(Controller...);
```
